samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/smp_svr.c

```diff
@@ -0,0 +1,73 @@
+/* Bluetooth: Mesh Generic OnOff, Generic Level, Lighting & Vendor Models
+ *
+ * Copyright (c) 2018 Vikrant More
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "common.h"
+
+#ifdef ENABLE_SMP_SVR
+
+#include <assert.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/gatt.h>
+#include <mgmt/buf.h>
+#include <mgmt/smp_bt.h>
+#include <stats.h>
+#include <stdlib.h>
+#include <string.h>
+#include <zephyr.h>
+
+#ifdef CONFIG_MCUMGR_CMD_IMG_MGMT
+#include "img_mgmt/img_mgmt.h"
+#endif
+#ifdef CONFIG_MCUMGR_CMD_OS_MGMT
+#include "os_mgmt/os_mgmt.h"
+#endif
+#ifdef CONFIG_MCUMGR_CMD_STAT_MGMT
+#include "stat_mgmt/stat_mgmt.h"
+#endif
+
+/* Define an example stats group; approximates seconds since boot. */
+STATS_SECT_START(smp_svr_stats)
+STATS_SECT_ENTRY(ticks)
+STATS_SECT_END;
+
+/* Assign a name to the `ticks` stat. */
+STATS_NAME_START(smp_svr_stats)
+STATS_NAME(smp_svr_stats, ticks)
+STATS_NAME_END(smp_svr_stats);
+
+/* Define an instance of the stats group. */
+STATS_SECT_DECL(smp_svr_stats) smp_svr_stats;
+
+void smp_svr_init(void)
+{
+	int rc;
+
+	rc = STATS_INIT_AND_REG(smp_svr_stats, STATS_SIZE_32, "smp_svr_stats");
+	assert(rc == 0);
+
+	/* Register the built-in mcumgr command handlers. */
+#ifdef CONFIG_MCUMGR_CMD_FS_MGMT
+	fs_mgmt_register_group();
+#endif
+#ifdef CONFIG_MCUMGR_CMD_OS_MGMT
+	os_mgmt_register_group();
+#endif
+#ifdef CONFIG_MCUMGR_CMD_IMG_MGMT
+	img_mgmt_register_group();
+#endif
+#ifdef CONFIG_MCUMGR_CMD_STAT_MGMT
+	stat_mgmt_register_group();
+#endif
+}
+
+void smp_svr(void)
+{
+	STATS_INC(smp_svr_stats, ticks);
```

Comment:
Just looking at this command, it doesn't seem like it can't be done on the system work queue. If this is just updating some counters, it shouldn't take much time or consume a lot of stack.

Reply:
It is as per samples/subsys/mgmt/mcumgr/smp_svr.

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/13135#discussion_r255399490